### DISCUSSION TITLE
Add minimal requirements — launching on 4 GBs of VRAM

### DISCUSSION THREAD

**Discussion by ClashSAN** [deforum-art/sd-webui-text2video#27] [Ideas]
(Windows) (NVIDIA: Ampere) 4gb makes 192x192 videos, which are not bad!

![2](https://user-images.githubusercontent.com/98228077/226567768-fc6bf371-0e75-4a92-b26d-ef2511903cf4.gif)
![3](https://user-images.githubusercontent.com/98228077/226567855-e121aba5-e231-4362-8dd0-64b4ba5b9528.gif)
![4](https://user-images.githubusercontent.com/98228077/226567939-98e269cd-222f-462c-a44c-fb62698d6ba5.gif)
![5](https://user-images.githubusercontent.com/98228077/226567949-883f9524-567b-4690-95f1-00d4d25b8456.gif)
![6](https://user-images.githubusercontent.com/98228077/226567977-28e5af8e-649a-47c6-a661-7a3240c0c0e4.gif)
![7](https://user-images.githubusercontent.com/98228077/226567996-68037992-ab71-43fe-98eb-f29780cf4982.gif)
![9](https://user-images.githubusercontent.com/98228077/226568045-fce88983-a232-470b-9af8-10472f12bb47.gif)

**Comment by hithereai** [comment]
Indeed not bad. Which minimal requirements you wanted to discuss?

**Reply by ClashSAN** [reply]
192x192 is very coherent, 128x128 (doesn't do anything.)
there is no current advantage between newer and older gpus except for gpus handicapped to run at full precision.

- current minimal requirements: All 6gb fp16 capable gpus should be able to generate 192x192 videos.

**Reply by michaelbogdan** [reply]
On my Nvidia 2060 Super with 8GiB of VRAM I got video generation with 24 frames and 320x320 resolution to work at half precision, 360x360 fails with out of memory error. So for that resolution I think this is the minimal requirement.

**Comment by kabachuha** [comment]
I have a GPU with 4 gb VRAM, I'll check if it works

**Reply by kabachuha** [reply]
Not really worked for me. And(!) I found initial SD-checkpoint unloading doesn't lower the vram usage at all for some reason (tested with a delay after the deletion), so something may be wrong

**Reply by ClashSAN** [reply]
@kabachuha I unloaded the model with that custom script, can you test it again?

```
import gradio as gr
import gc
from modules import sd_models,scripts,shared,sd_hijack,devices

class Script(scripts.Script):
    def title(self):
        return "Unload Button"

    def show(self, is_img2img):
        return True

    def ui(self, is_img2img):
        unloadmodel = gr.Button(value="unload model",variant='primary')

        def unload():
            sd_hijack.model_hijack.undo_hijack(shared.sd_model)
            shared.sd_model = None
            gc.collect()
            devices.torch_gc()
            return

        unloadmodel.click(fn=unload)
```


![image](https://user-images.githubusercontent.com/98228077/226756494-67302730-4959-4202-aa19-1abce9bc6a83.png)
![image](https://user-images.githubusercontent.com/98228077/226757483-935bc621-636e-46ef-ae81-02105e4a9f75.png)

**Reply by kabachuha** [reply]
I've accidentally deleted my webui folder with the models when trying to update it from scratch, so it's going to take a little while. (I suppose it's not as well updated as it should be)

The most important thing is that you've shown the GPU limitation screenshot, so I know that it *fits* within 4 gbs. (don't you have rtx 3050 as well incidentally?)

---

Pinned for more attention from testers

**Reply by kabachuha** [reply]
Idk why, but that button unloading still leaves a considerable amount of VRAM, which is preventing me from launch the model on your described resolution. Do you have any idea what module it could be?

![Screenshot_20230322_035358](https://user-images.githubusercontent.com/14872007/226774310-4d3a464a-00d0-43e5-a975-cbc1b9a0a34b.png)

**Reply by ClashSAN** [reply]
EDIT: @kabachuha the latest text2video_pytorch_model is the problem. If you use the text2video_pytorch_model from this commit,  you will have no issue. https://huggingface.co/damo-vilab/modelscope-damo-text-to-video-synthesis/tree/7ecc3083c7f3b4214f2dbf61cbf300281f0f7b67 

(Remove certain extensions.) I noticed the with aesthetic gradients extension active, the button did not allow me to unload the model completely. I was creating some recently.

 Running modelscope would continue until it would OOM during vae decode phase.

>Idk why, but that button unloading still leaves a considerable amount of VRAM, which is preventing me from launch the model on your described resolution. Do you have any idea what module it could be?

If I generate with sd, then unload, I see the residual vram as you have, but regardless of that I can still generate at 192 afterwards. I launched webui with a fp32 1.5 .ckpt but the others should work.
Try launching with --lowvram mode just ensuring it works for you at 192? It looks like this is possible with a 4gb 980 which is good to see: https://github.com/deforum-art/sd-webui-modelscope-text2video/issues/35

**Generate SD image, then do modelscope** \
\
![image](https://user-images.githubusercontent.com/98228077/226783089-471aad51-2321-46e3-b011-f495be9c8afc.png)

After modelscope
![image](https://user-images.githubusercontent.com/98228077/226783147-1f3057e5-e566-49c9-9b6d-fcea004662d7.png)

DRAM
![image](https://user-images.githubusercontent.com/98228077/226784485-638e9e9d-ff29-4a1a-956b-c51b8770b551.png)

**Reply by kabachuha** [reply]
Still not worked even with the previous model.

I suspect it might be that the model is not being loaded fully in the memory before doing 'half' on it. I'm pruning the models to fp16 right now and will write if it helped. Pruned weights will be nice to have anyway to save the community's bandwidth

https://huggingface.co/kabachuha/modelscope-damo-text2video-pruned-weights

---

Well, now, it at least got past the first stage and broke on diffusion. I guess, some progress 🙂

**Reply by ClashSAN** [reply]
I do many consecutive generations, to get a feel for prompts, its consistent for me, just running with no flags on a very fresh installation and 9ef779cca59734560c24cb3889705dc7b5a3d346, used all your pruned weights.

Do you have [Hardware Scheduling Disabled](https://github.com/AUTOMATIC1111/stable-diffusion-webui/discussions/3889)?
Testing with Windows 11 GeForce RTX 3050 Laptop GPU

other:
disable browser hardware acceleration
Go in nvidia control panel, 3d parameters, and change power profile to "maximum performance" (dunno if i did)

>>plugged in
>>Driver Version 31.0.15.2698

I have the bloated system compared to W10 and linux, so maybe its your gpu having slightly less vram somehow?

>Well, now, it at least got past the first stage and broke on diffusion. I guess, some progress 🙂

What do you mean? it loads the model, then it drops?

**Reply by kabachuha** [reply]
I have the just same GeForce RTX 3050 Laptop GPU. Maybe, it's my KDE Linux setup that's breaking the processes to protect my system from overload? 🤔

I'll try running it on Windows

**Comment by diStyApps** [comment]
Thank you, works on 970
![970txt2vid](https://user-images.githubusercontent.com/3084832/226961007-4cebc88d-6874-4bf3-ab96-caf1a33b7c39.png)

**Reply by kabachuha** [reply]
Omg

**Reply by diStyApps** [reply]
I didn't know github got 10mb limit of video, anyway a quick demo on 970 and 3080 on different resolution and lengths 
https://www.youtube.com/watch?v=2tHn7lTcGR4

**Comment by balisujohn** [comment]
I can confirm this works with the half-size pruned models on a 1070ti on Ubuntu 22.04 with 8 gigs of vram, in case anyone has a similar card.
![image](https://user-images.githubusercontent.com/20377292/227099271-e083bae2-5e36-46e6-9909-00a2e27c7cf5.png)

**Reply by kabachuha** [reply]
>  8 gigs of vram

Well, that is not low :)

The question was more about the minimal ones, such as only 4 GBs of VRAM

**Comment by JD1234JD1234** [comment]
i cant run 192x192 with 4GB vram 1050ti, 128x128 and 128x192 will run but give random/glitch patterns
